seatunnel-connectors-v2/connector-cdc/connector-cdc-postgres/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/postgres/source/reader/PostgresSourceFetchTaskContext.java

```diff
@@ -180,28 +180,29 @@ public void configure(SourceSplitBase sourceSplitBase) {
                 snapshotter.init(connectorConfig, offsetContext.asOffsetState(), slotInfo);
             }
 
-            SlotCreationResult slotCreatedInfo = null;
             if (snapshotter.shouldStream()) {
-                final boolean doSnapshot = snapshotter.shouldSnapshot();
-                createReplicationConnection(
-                        doSnapshot, connectorConfig.maxRetries(), connectorConfig.retryDelay());
+                if (sourceSplitBase.isIncrementalSplit() || slotInfo == null) {
+                    final boolean doSnapshot = snapshotter.shouldSnapshot();
+                    createReplicationConnection(
+                            doSnapshot, connectorConfig.maxRetries(), connectorConfig.retryDelay());
+                }
                 // we need to create the slot before we start streaming if it doesn't exist
                 // otherwise we can't stream back changes happening while the snapshot is taking
                 // place
                 if (slotInfo == null) {
                     try {
-                        slotCreatedInfo =
+                        SlotCreationResult slotCreatedInfo =
```

Comment:
Please remove useless variable definitions.

Reply:
fix by this pr
https://github.com/apache/seatunnel/pull/6785/files#diff-68a4e18adfe57dda2d37a3fc3092effa5c1a326076a16bb88131e82a62dba38fL193